runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -672,11 +706,84 @@ public void freeTableData(String namespace, String tableName) {
         table.clearAll();
     }
 
+    /**
+     * Clear the table by appending CLEAR to the stream. Start a transaction to mark the
+     * table as dropped in RegistryTable, and add this table and its checkpoint stream to the
+     * transaction's deleteSet, which instructs the sequencer to clean up the in-mem maps.
+     *
+     * @param namespace Namespace of the table.
+     * @param tableName Name of the table.
+     */
+    public void dropTable(String namespace, String tableName) {
+        log.info("Invoking dropTable {}${}", namespace, tableName);
+        clearAndUnregisterTable(namespace, tableName);
+
+        String fullyQualifiedTableName = getFullyQualifiedTableName(namespace, tableName);
+        tableMap.remove(fullyQualifiedTableName);
+        log.info("Done dropping table {}${}", namespace, tableName);
+    }
+
+    private void clearAndUnregisterTable(String namespace, String tableName) {
+        TableName tableNameKey = TableName.newBuilder()
+                .setNamespace(namespace)
+                .setTableName(tableName)
+                .build();
+
+        // Mark this table as dropped in RegistryTable
+        int numRetries = 9; // Since this is an internal transaction, retry a few times before giving up.
+        while (numRetries-- > 0) {
+            if (TransactionalContext.isInTransaction()) {
+                throw new IllegalThreadStateException("openTable: Called on an existing transaction");
+            }
+            try {
+                this.runtime.getObjectsView().TXBuild()
+                        .type(TransactionType.WRITE_AFTER_WRITE)
+                        .build()
+                        .begin();
+
+                // Clear table data
+                UUID streamId = CorfuRuntime.getStreamID(getFullyQualifiedTableName(namespace, tableName));
+                SMREntry entry = new SMREntry("clear", new Array[0], Serializers.PRIMITIVE);
+                TransactionalContext.getCurrentContext().logUpdate(streamId, entry);
+
+                UUID cpStreamId = CorfuRuntime.getCheckpointStreamIdFromId(streamId);
+                TransactionalContext.getCurrentContext().addToDeleteSet(streamId);
+                TransactionalContext.getCurrentContext().addToDeleteSet(cpStreamId);
+
+                CorfuRecord<TableDescriptors, TableMetadata> oldRecord =
+                        this.registryTable.get(tableNameKey);
+                if (oldRecord == null || oldRecord.getPayload() == null) {
+                    log.warn("The table to unregister {} does not exist in RegistryTable.", tableNameKey);
+                } else {
+                    CorfuRecord<TableDescriptors, TableMetadata> newRecord = new CorfuRecord<>(
+                            oldRecord.getPayload(),
+                            TableMetadata.newBuilder().mergeFrom(oldRecord.getMetadata()).setIsDropped(true).build());
+                    this.registryTable.insert(tableNameKey, newRecord);
```

Comment:
Maybe I missed something but when is the entry removed from the registryTable if we are attempting a soft delete here?
Also how does it work with openTable() which seems to be checking for absence of the entry itself as a pre-condition for dropping before open?

Reply:
1. I will work on that in a later PR. https://github.com/CorfuDB/CorfuDB/pull/3540#issuecomment-1487408937 I'm thinking to record the timestamp at which the table is marked as dropped, and based on this info let compactor clean up the entry.
2. A normal openTable() called on a table for the first time will pass the sequencer check. Only a table not existing in RegistryTable but exist in sequencer maps will be invalidated by sequencer. https://github.com/CorfuDB/CorfuDB/pull/3540/files/d45db7870d54786b944a43f54575eb22ae1cfa0f#diff-3ef8175cea450001f0ec6ce1f7f00df50a3626212a0c463cffcb464b94e95c1bR303